### ISSUE TITLE
Using customHash has returned to prepending multiple times again

### ISSUE THREAD

**Issue by typeoneerror** [rickharrison/broccoli-asset-rev#99]
(note: closed https://github.com/rickharrison/broccoli-asset-rewrite/issues/43 in favor of this issue)

Updated to latest and assets are now getting "assets" three times when using customHash and prepend.

in:
`<img src="assets/icons/stripe.svg">`
out:
`<img src="http://localhost:4200/assets/assets/assets/icons/stripe.svg">`

```
  var fingerprintOptions = {
    enabled: true,
    extensions: ['js', 'css', 'png', 'jpg', 'gif', 'map', 'svg'],
    generateAssetMap: true,
    prepend: 'http://localhost:4200/',
    customHash: null,
    generateAssetMap: false
  };
```

Looks specific to customHash again (removing that fixes again) and may be related to https://github.com/rickharrison/broccoli-asset-rev/issues/87

There already appears to be a failing test in master. "skips fingerprint and prepends when set and customHash is null" is prepending the url three times in the test and failing:

```diff
-  <script type="text/javascript" src="https://foobar.cloudfront.net/https://foobar.cloudfront.net/https://foobar.cloudfront.net/https://foobar.cloudfront.net/assets/application.js"></script>
-  <script type="text/javascript" src=https://foobar.cloudfront.net/https://foobar.cloudfront.net/https://foobar.cloudfront.net/https://foobar.cloudfront.net/assets/application.js></script>
+  <script type="text/javascript" src="https://foobar.cloudfront.net/assets/application.js"></script>
+  <script type="text/javascript" src=https://foobar.cloudfront.net/assets/application.js></script>
```

**Comment by typeoneerror** [comment]
FYI, the assetMap generated when `generateAssetMap: true` looks correct, so this may be an ember-cli issue.

**Comment by bichotll** [comment]
The same over here :/

**Comment by bichotll** [comment]
Hi @typeoneerror, any solution or at least workaround regarding this? ;)

I needed for a commercial project, I'm afraid :/

**Comment by typeoneerror** [comment]
@bichotll I didn't have any issues with production builds, it was only when I had `customHash: null` and `generateAssetMap: false` set in development build that it fails like this. So workaround is to just not use customHash at all locally and deal with the hash output on the dist files.

**Comment by rickharrison** [comment]
A failing test case would be really helpful in debugging this!

**Comment by grayt0r** [comment]
I ran into this issue and in the process of trying to create a failing test case realised, for me at least, that's it's actually caused by an open issue on `broccoli-asset-rewrite`: https://github.com/rickharrison/broccoli-asset-rewrite/issues/39.

We reference the same relative image URL a few times in our CSS and each of those references appears to be rewritten multiple times, resulting in `assets/` appearing in the path multiple times. This only happens when `customHash: null` is specified as if the file is fingerprinted then after the first rewrite the path will no longer match the fingerprint regex.

Happy to provide a failing test for the `broccoli-asset-rewrite` repo, but believe the test in the proposed PR to fix the issue linked above covers the same scenario.

**Comment by rickharrison** [comment]
Yes, that would be great if you had a failing test case over there.
